tests/test_validate_input_database.cc

```diff
@@ -0,0 +1,494 @@
+/* Copyright (c) 2021, the adamantine authors.
+ *
+ * This file is subject to the Modified BSD License and may not be distributed
+ * without copyright and license information. Please refer to the file LICENSE
+ * for the text and further information on this license.
+ */
+
+#define BOOST_TEST_MODULE validate_input_database
+
+#include <validate_input_database.hh>
+
+#include "main.cc"
+
+namespace adamantine
+{
+
+BOOST_AUTO_TEST_CASE(expected_passes)
+{
+  boost::property_tree::ptree database;
+
+  // Bare-bones required inputs
+  database.put("boundary.type", "adiabatic");
+  database.put("discretization.fe_degree", "1");
+  database.put("geometry.dim", "3");
+  database.put("discretization.quadrature", "gauss");
+  database.put("geometry.import_mesh", "false");
+  database.put("geometry.length", 1.0);
+  database.put("geometry.height", 1.0);
+  database.put("geometry.width", 1.0);
+  database.put("materials.n_materials", 1);
+  database.put("materials.property_format", "polynomial");
+  database.put("materials.material_0.solid.thermal_conductivity_x", 10.);
+  database.put("materials.material_0.solid.thermal_conductivity_y", 10.);
+  database.put("materials.material_0.solid.thermal_conductivity_z", 10.);
+  database.put("materials.material_0.solid.density", 10.);
+  database.put("materials.material_0.solid.specific_heat", 10.);
+  database.put("post_processor.filename_prefix", "output");
+  database.put("refinement.n_heat_refinements", 0);
+  database.put("sources.n_beams", 1);
+  database.put("sources.beam_0.type", "goldak");
+  database.put("sources.beam_0.scan_path_file", "sp.txt");
+  database.put("sources.beam_0.scan_path_file_format", "segment");
+  database.put("sources.beam_0.depth", 0.1);
+  database.put("sources.beam_0.absorption_efficiency", 0.1);
+  database.put("time_stepping.method", "forward_euler");
+  database.put("time_stepping.duration", 1.0);
+  database.put("time_stepping.time_step", 0.1);
+  validate_input_database(database);
+}
+
+BOOST_AUTO_TEST_CASE(parse_test_input_files)
+{
+  boost::property_tree::ptree database;
+
+  std::string filename = "bare_plate_L_da.info";
+  boost::property_tree::info_parser::read_info(filename, database);
+  validate_input_database(database);
+
+  filename = "bare_plate_L_ensemble.info";
+  boost::property_tree::info_parser::read_info(filename, database);
+  validate_input_database(database);
+
+  filename = "demo_316_short.info";
+  boost::property_tree::info_parser::read_info(filename, database);
+  validate_input_database(database);
+
+  filename = "integration_2D.info";
```

Comment:
This should be:
```suggestion
  filename = "integration_2d.info";
```
I am not sure why it passes in release mode.

Reply:
My bad it's failing in release mode too

Reply:
That's really odd/concerning that it worked in both debug and release mode on my computer...

Regardless, that typo is now fixed.

Reply:
Maybe the file existed on your computer. If you are on mac the filesystem is the file system is not case sensitive. If the file was mapped from the docker to the mac, maybe something weird happened.